packages/shared/src/toast/Toast.js

```diff
@@ -20,58 +25,85 @@ type State = {|
   +isVisible: boolean,
 |};
 
+type HandlerChangeEvent = {
+  +nativeEvent: {
+    +state: number,
+    ...
+  },
+  ...
+};
+
 export default class Toast extends React.Component<Props, State> {
-  opacityValue = new Animated.Value(0);
+  translateY = new Animated.Value(-250);
   state = {
     isVisible: false,
   };
 
+  timeout: ?TimeoutID;
   static defaultProps = {
     duration: 3000,
   };
 
   show = () => {
     this.setState({ isVisible: true }, () => {
-      Animated.timing(this.opacityValue, {
-        toValue: 1,
-        duration: 500,
+      Animated.timing(this.translateY, {
+        toValue: 0,
+        duration: 250,
         useNativeDriver: true,
       }).start(() => {
-        setTimeout(() => {
-          this.hide();
-        }, this.props.duration);
+        this.timeout = setTimeout(this.hide, this.props.duration);
       });
     });
   };
 
   hide = () => {
-    Animated.timing(this.opacityValue, {
-      toValue: 0,
+    Animated.timing(this.translateY, {
+      toValue: -250,
```

Comment:
Btw what if height of the toast will be bigger than 250px?

Reply:
Then I could always increase it. But it will be hidden when the animation ends anyways.